modules/openy_gc_storage/openy_gc_storage.post_update.php

```diff
@@ -90,3 +90,153 @@ function openy_gc_storage_post_update_migrate_eventseries_instructors(&$sandbox)
+    ],
+    [
+      'title' => '20 minutes',
+      'low' => 1081,
+      'high' => 1319,
+    ],
+    [
+      'title' => '30 minutes',
+      'low' => 1320,
+      'high' => 1919,
+    ],
+    [
+      'title' => '45 minutes',
+      'low' => 1920,
+      'high' => 2819,
+    ],
+    [
+      'title' => '60 minutes',
+      'low' => 2820,
+      'high' => 3899,
+    ],
+    [
+      'title' => '90 minutes',
+      'low' => 3900,
+      'high' => 5700,
+    ],
+    [
+      'title' => 'Undefined',
+      'low' => 0,
+      'high' => 0,
+    ],
+  ];
+
+  $term_storage = \Drupal::entityTypeManager()
+    ->getStorage('taxonomy_term');
+
+  foreach ($durations as $duration) {
+    $term = $term_storage->create([
+      'vid'                   => 'gc_duration',
+      'name'                  => $duration['title'],
+      'field_gc_duration_min' => $duration['low'],
+      'field_gc_duration_max' => $duration['high'],
+    ]);
+    $term->save();
+  }
+
+  $sandbox['#finished'] = 1;
+  return t('The Duration terms have been created.');
+}
+
+/**
+ * Helper method to build duration references based on duration field value.
+ */
+function _openy_gc_storage_build_duration_references(&$sandbox) {
+  if (!isset($sandbox['sandbox']['ids'])) {
+    $sandbox['sandbox']['ids'] = \Drupal::entityQuery('node')
+      ->condition('type', 'gc_video')
+      ->sort('nid')
+      ->execute();
+
+    // By default we'll set reference to 'Undefined' duration term.
+    $sandbox['sandbox']['default_duration_term'] = \Drupal::entityQuery('taxonomy_term')
+      ->condition('vid', 'gc_duration')
+      ->condition('field_gc_duration_min', 0)
+      ->condition('field_gc_duration_max', 0)
+      ->range(0, 1)
+      ->execute();
+    if (!empty($sandbox['sandbox']['default_duration_term'])) {
+      $sandbox['sandbox']['default_duration_term'] = reset($sandbox['sandbox']['default_duration_term']);
+    }
+    else {
+      $sandbox['sandbox']['default_duration_term'] = '';
+    }
+
+    $sandbox['sandbox']['max'] = count($sandbox['sandbox']['ids']);
+    $sandbox['sandbox']['processed'] = 0;
+  }
+
+  $node_storage = \Drupal::entityTypeManager()->getStorage('node');
+
+  for ($i = 0; $i < 5; $i++) {
+    if (empty($sandbox['sandbox']['ids'])) {
+      break;
+    }
+    $duration_term_tid = $sandbox['sandbox']['default_duration_term'];
+
+    $current_nid = array_shift($sandbox['sandbox']['ids']);
+    $node = $node_storage->load($current_nid);
+    /** @var \Drupal\node\NodeInterface $node */
+    $duration_field = $node->get('field_gc_video_duration');
+    // Let's find id of the duration term that has minimal value less than the
+    // duration of the given video & has maximum value bigger than or equal to
+    // the duration of the given video.
+    if (!$duration_field->isEmpty()) {
+      $duration_value = $duration_field->getString();
+      $duration_term = \Drupal::entityQuery('taxonomy_term')
+        ->condition('vid', 'gc_duration')
```

Comment:
It is better to get all duration items once before the first loop. it will save from reading that data in a loop.
In addition to that, we have to think about situations when the duration max is infinite (empty value or set max possible integer value).
At the current version of code durations like `45min+`